core/gesture.js

```diff
@@ -392,26 +410,55 @@ Blockly.Gesture.prototype.handleUp = function(e) {
 };
 
 /**
- * End the gesture associated with the given event, and remove it from the
- * gesture database.
- * @param {!Event} e A mouse move or touch move event.
+ * Cancel an in-progress gesture.  If a workspace or block drag is in progress,
+ * end the drag at the most recent location.
+ *
  */
-Blockly.Gesture.prototype.endGesture = function(e) {
+Blockly.Gesture.prototype.cancelGesture = function() {
+  Blockly.longStop_();
+  if (this.isDraggingBlock_) {
+    this.blockDragger_.endBlockDrag(this.mostRecentEvent_,
+        this.currentDragDeltaXY_);
+  } else if (this.isDraggingWorkspace_) {
+    this.workspaceDragger_.endDrag(this.currentDragDeltaXY_);
+  }
+  this.endGesture();
+};
+
+/**
+ * End the gesture associated and remove it from the gesture database.
+ * If an event is passed in, stop that event from doing anything else on the
+ * page.
+ * @param {Event} opt_e A mouse move or touch move event, if this gesture is
```

Comment:
Most of the time I'm ending a gesture based on a touchend/mouseup event.  But sometimes I want to cancel a gesture--that's the equivalent of terminateDrag.

For instance, zooming with the scroll wheel cancels in-progress drags so that we don't have to handle scale changing midway through a drag.  I don't want to muck around with the mousewheel event, I just want the gesture to clean up neatly.